### PR TITLE
feat: add new Drawer component

### LINKED ISSUE #7: Drawer
This is NOT for now but it is discussed to be introduced in the future.

image

We need to define a drawer inside the page maybe to pass content to it and show/hide it on demand.

Issue description underneath edited by @kouloumos

## Implementation
All the categories must have a transparent background underneath their icon that will be rounded. Only selected category will fill this circle with the branded1 color.
The icon when the branded1 color exists must be the color from the swatches based on that color
The subcategories will not have that rounded bg. Only Icons.
Subcategory when selected will have a bacgkround color (lighter gray).

## Design
The design can be found [here](https://www.figma.com/file/8kMPBNYHHXz2AtkzeeDmk5/Design-System?node-id=870%3A26958)

### PR DESCRIPTION
**Pull request by kouloumos**
## Description

This PR implements the Drawer as described in issue #7
Resolves #7

## Screenshot
Part of the documentation
Screenshot 2021-01-07 at 9 43 18 AM

## Test Plan

- Test that menu items are rendering
- Test that submenu items are displayed when menu item is clicked
- Test that submenu navigation is working

### CODE REVIEW

File: src/components/Drawer/Navigation/MenuItem/MenuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  isCurrent: boolean;
+  /** Defines if the menu item is expanded */
+  expanded: boolean;
+  toggleMenuItem: (newUrl: string) => void;
+} & MenuItemProps;
+
+const MenuItem: React.FC<Props> = memo(
+  ({ isCurrent, expanded, name, url, iconName, options, toggleMenuItem }) => {
+    const { calculateColorBetweenColorAndType } = useTypeColorToColorMatch();
+    const branded1Color = calculateColorBetweenColorAndType('', 'branded1');
```
Comment (panvourtsis) [inline src/components/Drawer/Navigation/MenuItem/MenuItem.tsx:31]: these two lines could be memoized

File: src/components/Drawer/Navigation/MenuItem/MenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    const calculatedIconColor = pickTextColorFromSwatches(branded1Color.color, branded1Color.shade);
+
+    const hasSubMenus = options.length > 0;
```
Comment (panvourtsis) [inline src/components/Drawer/Navigation/MenuItem/MenuItem.tsx:34]: this could be memoized

File: src/components/Drawer/Navigation/Navigation.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        return newOpenMenuItems;
+      }
+    });
+  }, []);
+
+  // we need to show an indication on the currently selected option
+  // we get that information from the current URL and then comparing it with the urls of each menuItem
+  useLayoutEffect(() => {
+    menuItems.some(menuItem => {
```
Comment (panvourtsis) [inline src/components/Drawer/Navigation/Navigation.tsx:34]: use of `.some` indicates that will return something in a const for something else. If this is just a loop `.forEach` could work just fine

File: src/components/storyUtils/DrawerShowcase/MenuItems.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import { MenuItem } from '../../Drawer/types';
```
Comment (panvourtsis) [inline src/components/storyUtils/DrawerShowcase/MenuItems.tsx:1]: this file is `.ts`. The extension `.tsx` indicates jsx usage somewhere inside

**Comment by panvourtsis** [comment]
another small fix is the alignment
image
vs
image

**Review by panvourtsis — APPROVED** [review]

File: src/components/Drawer/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        return newOpenMenuItems;
+      }
+    });
+  }, []);
+
+  // we need to show an indication on the currently selected option
+  // we get that information from the current URL and then comparing it with the urls of each menuItem
+  useLayoutEffect(() => {
+    menuItems.forEach(menuItem => {
+      const menuItemOptions = menuItem.options.map(subMenuItem => subMenuItem.url);
```
Comment (michalisKout) [inline src/components/Drawer/Navigation/Navigation.tsx:35]: Is this a list of menuItemUrls instead?

File: src/components/Drawer/Navigation/Navigation.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        newOpenMenuItems.splice(menuItemIndex, 1);
+
+        return newOpenMenuItems;
+      }
+    });
+  }, []);
+
+  // we need to show an indication on the currently selected option
+  // we get that information from the current URL and then comparing it with the urls of each menuItem
+  useLayoutEffect(() => {
```
Comment (michalisKout) [inline src/components/Drawer/Navigation/Navigation.tsx:33]: Optional: This hook alongside with useLocation hook can be moved into a custom hook.

**Review by michalisKout — APPROVED** [review]

File: src/components/Drawer/Navigation/Navigation.tsx (hunk 5 above)
Comment (kouloumos) [reply src/components/Drawer/Navigation/Navigation.tsx:35]: yes it is

File: src/components/Drawer/Navigation/Navigation.tsx (hunk 6 above)
Comment (kouloumos) [reply src/components/Drawer/Navigation/Navigation.tsx:33]: is there a reason for this if this hook will not be used anywhere else?

File: src/components/Drawer/Navigation/Navigation.tsx (hunk 5 above)
Comment (michalisKout) [reply src/components/Drawer/Navigation/Navigation.tsx:35]: then we can rename it to something related to its value 😃

File: src/components/Drawer/Navigation/Navigation.tsx (hunk 6 above)
Comment (michalisKout) [reply src/components/Drawer/Navigation/Navigation.tsx:33]: It's an optional change. 
Reasons: logic extraction and readability improvement

Comment (kouloumos) [reply src/components/Drawer/Navigation/Navigation.tsx]: done

**Review by michalisKout — APPROVED** [review]

**Pull request merged**